ksql-engine/src/main/java/io/confluent/ksql/query/id/SpecificQueryIdGenerator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.query.id;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Returns a specific query Id identifier based on what's set. Only returns each set Id once and
+ * will throw an exception if getNextId() is called twice without update.
+ */
+public class SpecificQueryIdGenerator implements QueryIdGenerator {
+
+  private long nextId;
+  private boolean alreadyUsed;
+
+  public SpecificQueryIdGenerator() {
+    this.nextId = 0L;
+    this.alreadyUsed = true;
+  }
+
+  public void setNextId(final long nextId) {
+    alreadyUsed = false;
+    this.nextId = nextId;
+  }
+
+  @Override
+  public long peekNext() {
+    return alreadyUsed ? nextId + 1 : nextId;
+  }
+
+  @Override
+  public String getNext() {
+    if (alreadyUsed) {
+      throw new NoSuchElementException("QueryIdGenerator has not been updated with new offset");
+    }
+
+    alreadyUsed = true;
+    return String.valueOf(nextId);
+  }
+
+  @Override
+  public QueryIdGenerator createSandbox() {
+    // We don't expect to have this function called for this type of generator
```

Comment:
🤔 I think here we can just return a `SequentialQueryIdGenerator` with the current nextId as the seed - if it's a sandbox then it should be fine and at least we implement the correct APIs

Reply:
That won't work for batched requests because we need to generate multiple IDs from the sandbox. I think returning a sequential generator from the current ID is fine for the sandbox case.

Reply:
> That won't work for batched requests 
What won't work?

Reply:
As it stands right now, it won't work, it would require modifying RequestValidator to call setNextId()
https://github.com/confluentinc/ksql/pull/3354#discussion_r327756457

Reply:
Ah I missed that you said `SequentialQueryIdGenerator`. Disregard.